data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/codec/JsonCodecTest.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.codec;
+
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+class JsonCodecTest {
+
+    @ParameterizedTest
+    @ValueSource(ints = { 2 })
+    void parse_with_events_outputstream_json_codec(final int numberOfObjects) throws IOException {
+
+        final Map<String, String> eventData = new HashMap<>();
+        eventData.put("key1", "value");
+        eventData.put("key2", "value");
+        final JacksonEvent event = JacksonLog.builder().withData(eventData).withEventType("LOG").build();
+        String output = createObjectUnderTest().parse(event);
+        assertNotNull(output);

```

Comment:
Please also verify the actual object.

I think the best way in this scenario is to have an `ObjectMapper` in the test. Then you can deserialize the `output` and verify it has the same data.

```
Map deserializedData = objectMapper.readValue(output, Map.class);
assertThat(deserializedData.get("key1"), equalTo(value1));
assertThat(deserializedData..get("key2"), equalTo(value2));
// maybe some more assertions?
```

Reply:
Resolved and added more assertions.